### PR TITLE
add code splitting support

### LINKED ISSUE #42: Code splitting for spesific service
We know firebase bundle is too big so, I think we can add support for code splitting. I have try to add code splitting with async/await in `plugin.js` and its works. You can check the pr

### PR DESCRIPTION
**Pull request by fachrihawari**
It will add support for code splitting and will resolve #42 issue

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -18,50 +10,66 @@ export default (ctx, inject) => {
...
   }
 
   // Firebase Messaging can only be initiated on client side
-  if (process.browser && options.useOnly.includes('messaging') && firebase.messaging.isSupported()) {
-    const fireMess = firebase.messaging()
-    const fireMessObj = firebase.messaging
-    inject('fireMess', fireMess)
-    inject('fireMessObj', fireMessObj)
+  if (process.browser && options.useOnly.includes('messaging')) {
+    <%= options.useOnly.includes('firestore') ? "await import('firebase/firestore')" : "" %>
```
Comment (lupas) [inline plugin.js]: should be 
```
<%= options.useOnly.includes('messaging') ? "await import('firebase/messaging')" : "" %>
```

**Comment by lupas** [comment]
Hey @fachrihawari 

Thanks a lot for the pull request! :)

I did a nuxt build analyze test for [nuxt-fire-demo](https://github.com/lupas/nuxt-fire-demo) and got the following results:

- Bundle size stays almost the same:
295.44 KB (without PR)
299.67 KB (with PR)

- However, Firebase modules get taken out of vendor.app.js and splitted into separate files:

Without PR:


With PR:


So that seems to be a good thing, vendor.app.js get's much smaller.

I'm not a pro when it comes to optimizing js bundles however, so just 3 shy questions before we'll merge that :)

1. Do you know the reason behind why dynamic imports splits the modules into separate files, whereas the standard lodash conditional imports at the top don't?

2. Does the split in separate files greatly improve page load speeds with benefits for the end users?

3. Since we are currently NOT compiling the module and we are using dynamic imports here, which were introduced in ES6, is it guaranteed that this will work with all setups & older browsers? 

Do you maybe have any (e.g. StackOverflow) links where this method is suggested to improve Firebase's performance (for regular, not nuxt-fire, setups)?

Looks really promising, would be great if that would improve the performance of nuxt-fire! :) 

PS: Oh and line 59 needs fixing, just a small copy paste mistake I assume.

**Comment by fachrihawari** [comment]
Hi @lupas 
Happy to see your response. The reason why I made this PR is for make the service loaded dynamically not on the first time load.

1. dynamic import introduce for js module like in this [link](https://v8.dev/features/dynamic-import). Why the dynamic split our code to separated file because is handled by babel and web pack for splitting code which use dynamic import

2. Of course yes, because splitted file will not blocking the rendering of page. You can see this [video](https://www.youtube.com/watch?v=DHbVyRLkX4c) from firebase summit 2019

3. From the video above, its can handle with polyfill but I'm not yet test.

File: plugin.js (hunk 1 above)
Comment (fachrihawari) [reply plugin.js]: Ah, I am so excited about this until I rush on the code 😅

**Comment by lupas** [comment]
@fachrihawari 

Thank you for the detailed information! This is great!

Everything looks fine. If support for older browsers is needed and it would break because of this, there is always the option to add the import() polyfill.

Thank you for your input!

**Comment by lupas** [comment]
Implemented with [v2.2.0](https://github.com/lupas/nuxt-fire/releases/tag/v2.2.0).

**Comment by fachrihawari** [comment]
@lupas waw, its great :hooray:

**Pull request merged**